### PR TITLE
Some enhancements

### PR DESCRIPTION
**Pull request by KostyaSha**

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java [hunk 1]
```diff
@@ -141,27 +142,31 @@ public File getLogFile() {
...
-                                if(createJenkinsHook(repo, getDescriptor().getHookUrl())) {
-                                    continue OUTER;
-                                }
-                            } catch (Throwable e) {
-                                LOGGER.log(Level.WARNING, "Failed to add GitHub webhook for "+name, e);
+            registerHooks();
+        }
+    }
+
+    public void registerHooks() {
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: Do you really need to keep it public? If yes, Javadoc is required

Comment (KostyaSha) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: Commit description says "Move to separate method to have ability re-register hooks" from groovy for example.

Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: Yes, My bad. I finally bothered to read the PR

File: src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java [hunk 2]
```diff
@@ -320,6 +326,25 @@ public FormValidation doCheckHookUrl(@QueryParameter String value) {
 
         }
 
+        public FormValidation doReRegister() {
+            if (!manageHook) {
+                return FormValidation.error("Works only when jenkins manages hooks");
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: **J**enkins

File: src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java [hunk 3]
```diff
@@ -320,6 +326,25 @@ public FormValidation doCheckHookUrl(@QueryParameter String value) {
...
+            }
+
+            int triggered = 0;
+            for (AbstractProject<?,?> job : Jenkins.getInstance().getAllItems(AbstractProject.class)) {
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: Check for null?
Somebody can call the method before the full startup

File: src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java [hunk 4]
```diff
@@ -320,6 +326,25 @@ public FormValidation doCheckHookUrl(@QueryParameter String value) {
...
+                GitHubPushTrigger trigger = job.getTrigger(GitHubPushTrigger.class);
+                if (trigger!=null) {
+                    LOGGER.log(Level.FINE, "Calling registerHooks() for {0}", job.getFullName());
+                    trigger.registerHooks();
+                    triggered++;
+                }
+            }
+
+            LOGGER.log(Level.INFO, "Called registerHooks() for {0} jobs", triggered);
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: I would recommend to have something human-readable. Jenkins users do not know what <code>registerHooks()</code> means

Comment (KostyaSha) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: This is log, it also has different hashes for objects, weird class names and etc.

File: src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java [hunk 5]
```diff
@@ -228,6 +233,7 @@ public void writeLogTo(XMLOutput out) throws IOException {
 
     @Extension
     public static class DescriptorImpl extends TriggerDescriptor {
+        private static final Logger LOGGER = Logger.getLogger(DescriptorImpl.class.getName());
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: I would recommend a logger for <code>GitHubPushTrigger.class</code>. Such approach would be much more useful for logs filtering

File: src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java (hunk 3 above)
Comment (KostyaSha) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: Like in other code.

File: src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java (hunk 5 above)
Comment (KostyaSha) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: I need see log from Descriptor and not from multiple triggers. Descriptor is singleton and i don't need to track from what Describable instance this log was printed.

File: src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java [hunk 6]
```diff
@@ -141,27 +142,35 @@ public File getLogFile() {
...
-                            } catch (Throwable e) {
-                                LOGGER.log(Level.WARNING, "Failed to add GitHub webhook for "+name, e);
+            registerHooks();
+        }
+    }
+
+    /**
+     * Tries to register hook for current associated job.
+     * Useful for using from groovy scripts.
+     */
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java]: @since TODO

**Comment by KostyaSha** [comment]
rebased, force-pushed

**Comment by KostyaSha** [comment]
User confirmed that it works

**Pull request merged**